### PR TITLE
Update annotations by using old features

### LINKED ISSUE #660: Error after splitting video annotation with new label
I found the following error in the video annotation tool (coming from [here](https://github.com/biigle/core/blob/8f9c0864c54bb7ee57d2a910f0ce7ba1bccafbe1/resources/assets/js/videos/components/videoScreen/annotationPlayback.vue#L96)):

```
Error in event handler for "refresh": "TypeError: this.featureChangeKeys_[featureKey] is undefined"
```

Steps to reproduce:

1. Create a label tree with at least two labels (or use an existing one)
2. Create a point video annotation
3. Split the annotation in the middle
4. Use the attach label interaction to add the second label the first part of the annotation
5. Delete the first label from the part of the video annotation
6. Move the current time of the video to be inside the first part of the annotation
7. Play the video

The error occurs when the current time of the video jumps from the first part of the annotation to the second.

### PR DESCRIPTION
**Pull request by lehecht**
Fix #660

### CODE REVIEW

**Comment by mzur** [comment]
Can you please elaborate why this fixes the original issue (as the reported error happens somewhere else)?

**Comment by lehecht** [comment]
I fixed a very similar error [here](https://github.com/biigle/core/blob/c640be10f94177962385d578d214278422ac5f12/resources/assets/js/annotations/components/annotationCanvas.vue#L523-L528).
The error occurs, because you try to delete an annotation, which does not exist any more in the annotation source. That's why you need to check for its presence. Some annotations are still present and can be deleted there, but others are already deleted. Moreover the method `hasFeature` in `source.hasFeature(feature)` cannot be used, because it still uses the old array of the source, which is why I needed to use `getFeatures().includes()`.

File: resources/assets/js/videos/components/videoScreen/annotationPlayback.vue [hunk 1]
```diff
@@ -90,10 +90,13 @@ export default {
                     toCreate.push(annotation.self);
                 }
             }
-
+
```
Comment (mzur) [inline resources/assets/js/videos/components/videoScreen/annotationPlayback.vue:93]:
```suggestion

```

**Review by mzur — COMMENTED** [review]
Ok thanks. I can confirm that the error is gone with the change. However, I see a visual bug where suddenly a second point appears on the video when the annotation label is deleted. Here is a recording (don't mind the "invisible video" bug of Firefox):

[Screencast from 04.01.2024 09:39:57.webm](https://github.com/biigle/core/assets/2457311/ae0335cc-a242-468e-b6be-8c121c586763)

The point is gone once the video played past the annotation(s). It also does not appear again if I rewind the video to the annotation.

**Comment by lehecht** [comment]
A little more explanation here.
My previous solution fixed the error only if the first annotation was selected. So the bug still existed when no annotation was selected.
I found out, that there were some problems with new created features and still existing one in the annotation source. The annotation source uses the featureChangeKeys method. For features not coming from the source directly the return value is always undefined, because they do not have the same keys like their predecessor(-copies).

File: resources/assets/js/videos/components/videoScreen/annotationPlayback.vue [hunk 2]
```diff
@@ -135,7 +136,13 @@ export default {
             source.addFeature(newFeature);
         },
         createFeature(annotation) {
```
Comment (mzur) [inline resources/assets/js/videos/components/videoScreen/annotationPlayback.vue:138]: In which case is `createFeature()` called with an annotation that already exists as a feature? I feel that the method should not be called in the first place in such a case.

File: resources/assets/js/videos/components/videoScreen/annotationPlayback.vue [hunk 3]
```diff
@@ -128,11 +128,12 @@ export default {
...
-            let oldFeature = source.getFeatureById(annotation.id)
-            
-            source.removeFeature(oldFeature);
-            source.addFeature(newFeature);
+            let feature = source.getFeatureById(annotation.id)
+
+            source.removeFeature(feature);
+
+            feature.set('color', annotation.labels[0].label.color);
+            source.addFeature(feature);
```
Comment (mzur) [inline resources/assets/js/videos/components/videoScreen/annotationPlayback.vue:136]: Looks good! Is the call to `removeFeature()` and `addFeature()` even necessary now?

**Review by mzur — APPROVED** [review]
Thanks!

**Pull request merged**